lib/utils.js

```diff
@@ -199,6 +199,9 @@ exports.encode = function encode (value) {
   else if (typeof value === 'string') {
     return '"' + exports.escape(value) + '"';
   }
+  else if (Object.prototype.toString.call(value) === '[object Date]') {
```

Comment:
I'd just use `value instanceof Date` here, I don't think it's worthwhile supporting cross-realm instances and this is quite a hot function so it will make a slight performance impact.